test/integration/good/code-gen/mother.stan

```diff
@@ -287,12 +300,59 @@ transformed parameters {
   cholesky_factor_cov[3] tp_cfcov_33;
   cholesky_factor_cov[3] tp_cfcov_33_ar[K];
 
+  tp_real_1d_ar = p_real_1d_ar;
+  tp_real_3d_ar = p_real_3d_ar;
+  tp_1d_vec = p_1d_vec;
+  tp_3d_vec = p_3d_vec;
+
+  tp_simplex = p_simplex;
+  tp_1d_simplex = p_1d_simplex;
+  tp_3d_simplex = p_3d_simplex;
+
+  tp_cfcov_54 = p_cfcov_54;
+  tp_cfcov_33 = p_cfcov_33;
+  tp_cfcov_33_ar = p_cfcov_33_ar;
+
+  for (i in 1:2) {
+    for (j in 1:3) {
+      for (m in 1:4) {
+        for (n in 1:5) {
+          tp_ar_mat[m, n, i, j] = 0.4;}}}}
```

Comment:
What about also testing the other kinds of indexing like
`a[x: ]`, `a[:x]`, `a[x:y]`, `a[:, 34]`, `a[{13,2,1}, :]`?
We'd ideally want to test them on both the lhs and rhs.